articles/architecture-scenarios/application/mobile-api.md

```diff
@@ -9,30 +9,369 @@ beta: true
+- __redirect_uri__: The URL must match the `redirect_uri` passed in the previous section to `/authorize`.
+
+The response from the Token URL will contain:
+
+```json
+{
+  "access_token": "eyJz93a...k4laUWw",
+  "refresh_token": "GEbRxBN...edjnXbL",
+  "id_token": "eyJ0XAi...4faeEoQ",
+  "token_type": "Bearer",
+  "expires_in":86400
+}
+```
+
+- __access_token__: An `access_token` for the API, specified by the `audience`.
+- __refresh_token__: A [Refresh Token](/tokens/refresh-token/current) will only be present if you included the `offline_access` scope AND enabled __Allow Offline Access__ for your API in the Dashboard.
+- __id_token__: An `id_token` JWT containing user profile information.
+- __token_type__: A string containing the type of token, this will always be a Bearer token.
+- __expires_in__: The amount of seconds until the `access_token` expires.
+
+You will need to store the above credentials in local storage for use in calling your API and retrieving the user profile.
+
+::: note
+[See the implementation in Android at the bottom of this document](/architecture-scenarios/application/mobile-api/mobile-implementation-android#store-credentials)
+:::
 
-- The `access_token` is used to securely call the API on behalf of the user.
+#### Get the User Profile
+
+To retrieve the [User Profile](/api/authentication?http#user-profile), your mobile application can decode the [ID Token](/tokens/id-token) using one of the [JWT libraries](https://jwt.io/#libraries-io). This is done by [verifying the signature](/tokens/id-token#verify-the-signature) and [validating the claims](/tokens/id-token#validate-the-claims) of the token. After validating the ID token, you can access its payload containing the user information:
+
+```json
+{
+  "email_verified": false,
+  "email": "test.account@userinfo.com",
+  "clientID": "q2hnj2iu...",
+  "updated_at": "2016-12-05T15:15:40.545Z",
+  "name": "test.account@userinfo.com",
+  "picture": "https://s.gravatar.com/avatar/dummy.png",
+  "user_id": "auth0|58454...",
+  "nickname": "test.account",
+  "created_at": "2016-12-05T11:16:59.640Z",
+  "sub": "auth0|58454..."
+}
+```
+
+::: note
+[See the implementation in Android at the bottom of this document](/architecture-scenarios/application/mobile-api/mobile-implementation-android#3-get-the-user-profile)
+:::
+
+#### Call the API
+
+To access secured resources from your API, the authenticated user's `access_token` needs to be included in requests that are sent to it. This is accomplished by sending the `access_token` in an `Authorization` header using the `Bearer` scheme.
+
+::: note
+[See the implementation in Android at the bottom of this document](/architecture-scenarios/application/mobile-api/mobile-implementation-android#4-call-the-api)
+:::
+
+#### Renew the Token
+
+To refresh your `access_token`, perform a `POST` request to the `/oauth/token` endpoint using the `refresh_token` from your authorization result. Your request should include:
```

Comment:
In the above `Get Credentials` section the `offline_access` scope and `Allow Offline Access` requirements are stated but it's probably a good idea to mention it again in this section.